### PR TITLE
Rebuild tx index index

### PR DESCRIPTION
**Pull request by m4ksio**
A small utility which populated missing entries in (block_id, tx_index) which was added mid-spork for mainnet16/testnet33.
I have run it on snapshot of data for mainnet16 and it worked fine.

New version started writing new index with deployment so going forward the data is there, and it will be there in next sporks.
We only miss entries from beginning of a spork up until deployment, so it's a one off operation.

While reviews are required, not sure if its worth merging to master after we run it current networks.

### CODE REVIEW

File: cmd/util/cmd/read-badger/cmd/transactions.go [hunk 1]
```diff
@@ -39,3 +53,162 @@ var transactionsCmd = &cobra.Command{
...
+
+				totalTxs++
+				txIndex++
+			}
+
+			blockNo++
+
+			if blockNo%10000 == 0 {
+				log.Info().Int("blocks", blockNo).Int("total_tx", totalTxs).Msg("processing progress")
+			}
```
Comment (koko1123) [inline cmd/util/cmd/read-badger/cmd/transactions.go:189]: What dies this check do? it is just logging for every 10k blocks?

**Review by koko1123 — APPROVED** [review]
looks fine to me, I'm assuming you'll get the lint issues :)

File: cmd/util/cmd/read-badger/cmd/transactions.go (hunk 1 above)
Comment (m4ksio) [reply cmd/util/cmd/read-badger/cmd/transactions.go:189]: Yes, it takes quite long to process (20-30mins)

File: cmd/util/cmd/read-badger/cmd/transactions.go [hunk 2]
```diff
@@ -39,3 +54,162 @@ var transactionsCmd = &cobra.Command{
...
+				if _, has := megaMap[txID][blockID]; has {
+					panic(fmt.Sprintf("duplicated tx %s in block %s", txID.String(), blockID.String()))
+				}
+
+				megaMap[txID][blockID] = struct{}{}
+
+				_, err := storages.TransactionResults.ByBlockIDTransactionIndex(blockID, txIndex)
+				if err != nil {
+					if errors.Is(err, storage.ErrNotFound) {
+						transactionResult, err := storages.TransactionResults.ByBlockIDTransactionID(blockID, txID)
```
Comment (peterargue) [inline cmd/util/cmd/read-badger/cmd/transactions.go:160]: should this report the error?

Comment (m4ksio) [reply cmd/util/cmd/read-badger/cmd/transactions.go:160]: No. It was previously, but turns out not every block received gets executed, so there might not be a tx result for given block and that's OK.
That's why I added a check for heights - there should be at least one tx for every height.

**Review by peterargue — APPROVED** [review]
Looks good. Added one comment about a missing error check. generally could use some more comments to make it easier to understand the flow, but it's probably only going to be used once

File: cmd/util/cmd/read-badger/cmd/transactions.go (hunk 2 above)
Comment (m4ksio) [reply cmd/util/cmd/read-badger/cmd/transactions.go:160]: Yep, record not found we can skip, but some other problems should bubble up, good catch

**Pull request closed without merging**